Source/Core/SwipeActions.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  Swipe.swift
+//  Eureka
+//
+//  Created by Marco Betschart on 14.06.17.
+//  Copyright © 2017 Xmartlabs. All rights reserved.
+//
+
+import Foundation
+
+public typealias SwipeActionHandler = (ContextualAction, ContextualActionSource, ((Bool) -> Void)?) -> Void
+
+public class SwipeAction{
+	public let contextualAction: ContextualAction
+	
+	@available(iOS 11, *)
+	public var backgroundColor: UIColor?{
+		get{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return nil }
+			return contextualAction.backgroundColor
+		}
+		set{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return }
+			contextualAction.backgroundColor = newValue
+		}
+	}
+	
+	@available(iOS 11, *)
+	public var image: UIImage?{
+		get{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return nil }
+			return contextualAction.image
+		}
+		set{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return }
+			return contextualAction.image = newValue
+		}
+	}
+	
+	public var title: String?{
```

Comment:
remove it since we can do `contextualAction.title`, `contextualAction.image` and `contextualAction. backgroundColor` and so on can be accessed directly using the protocol containing these properties.

Reply:
Despite its true these properties can be accessed directly it looks a bit awkward in the form definition code. That's why I also made `SwipeAction` conforming to the newly created `ContextualAction` protocol - and let `SwipeAction.contextualAction` create the platform specific action on demand.

This way we end up with a nice and clean looking form initialisation code:

```
let infoAction = SwipeAction(style: .normal, title: "Info", handler: { (action, row, completionHandler) in
	print("Info")
	completionHandler?(true)
})
infoAction.backgroundColor = .blue

$0.leadingSwipe.actions = [infoAction]
$0.leadingSwipe.performsFirstActionWithFullSwipe = true
```